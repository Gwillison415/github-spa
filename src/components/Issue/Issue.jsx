import React from 'react';
import './Issue.css'
import { Item, List, Label } from 'semantic-ui-react';
import { connect } from 'react-redux';

export const Issue = ({ title, repoIssueNumber, assignedTo, labels, issueId }) => {
  const renderedLabels = labels.map((label, i) => (
    <div>
      <Label.Group tag key={i}>
        <div className='us tiny label'>
          <Label>{label} </Label>
      </div></Label.Group>
    </div>
  )
)
  // const assignee = assignedTo.map((assignee) => (
  //   if(assignee.length===0) {
  //     return (
  //       <div></div>
  //     )
  //   } else {
  //     return (
  //       <div>
  //         <List.Description>Assigned To:{assignee}</List.Description>
  //       </div>
  //     )
  //   }
  // ))

    return (
      <List.Item textAlign='center'>
          <List.Content key ={issueId + 1 }className="ui center aligned">
            <List.Header>#{repoIssueNumber} {title}</List.Header>{renderedLabels}
            <List.Description>Assigned To: {assignedTo}</List.Description>
          </List.Content>
        </List.Item>
      )

}

export const mapStateToProps = (state, { issueId }) => {
  const issue = state.issues.issuesById[issueId];
  const { title, repoIssueNumber, assignedTo, labels  } = issue;
  return {
    title,
    repoIssueNumber,
    assignedTo,
    labels
  }
}

export default connect(
  mapStateToProps
)(Issue);
