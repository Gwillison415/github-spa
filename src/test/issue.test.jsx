import React from 'react';
import toJson, { shallowToJson } from 'enzyme-to-json';
import { shallow, render, mount } from 'enzyme';
import { Issue, mapStateToProps, mapDispatchToProps } from '../components/Issue/Issue';

const state = {

  issues: {
    issuesById: {
      238923429: {
        assignedTo: [['Gwillison415', 'https://avatars2.githubusercontent.com/u/22106099?v=4']],
        labels: ['duplicate', 'help wanted', 'question'],
        number: 16,
        title: 'login tst JWT token',
      },
    },
  },

};

describe('Issue entry component', () => {
  it('should render a component with props as specified ', () => {
    const component = shallow(

      <Issue
        labels={state.issues.issuesById[238923429].labels}
        issueId={238923429}
        assignedTo={[['Gwillison415', 'https://avatars2.githubusercontent.com/u/22106099?v=4']]}
        number={16}
        title={'login tst JWT token'}
        getDescription={() => { getDescription; }}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  it('map\'s given State To Props ', () => {
    const expected = {
      assignedTo: [['Gwillison415', 'https://avatars2.githubusercontent.com/u/22106099?v=4']],
      labels: ['duplicate', 'help wanted', 'question'],
      number: 16,
      title: 'login tst JWT token',
    };
    expect(mapStateToProps(state, { issueId: 238923429 })).toEqual(expected);
  });

  it('Should have a Cart Header in the center', () => {
    const component = shallow(
      <Issue
        number={16}
        title={'login tst JWT token'}
        issueId={238923429}
        labels={state.issues.issuesById[238923429].labels}
        assignedTo={[['Gwillison415', 'https://avatars2.githubusercontent.com/u/22106099?v=4']]}
      />,
    );
    expect(component.find('.aligned').exists()).toBe(true);
  });
});
