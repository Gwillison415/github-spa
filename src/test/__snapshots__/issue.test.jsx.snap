// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue entry component should render a component with props as specified  1`] = `
<ListItem
  textAlign="center"
>
  <ListContent
    className="ui center aligned"
  >
    <ListHeader>
      #
      16
       
      login tst JWT token
    </ListHeader>
    <div>
      <LabelGroup
        tag={true}
      >
        <div
          className="us tiny label"
        >
          <Label>
            duplicate
             
          </Label>
        </div>
      </LabelGroup>
    </div>
    <div>
      <LabelGroup
        tag={true}
      >
        <div
          className="us tiny label"
        >
          <Label>
            help wanted
             
          </Label>
        </div>
      </LabelGroup>
    </div>
    <div>
      <LabelGroup
        tag={true}
      >
        <div
          className="us tiny label"
        >
          <Label>
            question
             
          </Label>
        </div>
      </LabelGroup>
    </div>
    <ListDescription>
      Assigned To: 
      grant
    </ListDescription>
  </ListContent>
</ListItem>
`;
